README.md

```diff
@@ -46,12 +46,128 @@ The `main` and `db` packages contain integration tests that require a successful
+```
+{
+    domain: "example.com",
+    scandata: {
+        status: 0,
+        results: {  // Individual hostname check results
+            "mx.example.com": {
+                "status": 0,
+                "checks": {
+                    "connectivity": { "status": 0 },
+                    "certificate": { "status": 0 },
+                    "starttls": { "status": 0 },
+                    "version": { "status": 0 },
+                }
+            }
+            "dummy.example.com": {
+                "status": 3,
+                "checks": {
+                    "connectivity": {
+                        "status": 3,
+                        "messages": [ "Error: Could not establish connection" ]
+                    },
+                }
+            },
+        },
+        preferred_hostnames: ["mx.example.com"], // Hostnames we were able to connect to
+        extra_results: {"policylist": { "status": 0 }},
+    },
+    timestamp: 0,
+    version: 1,
+}
+```
+
+The meat of the response is in scandata, which is a JSON-ification of the `DomainResult` structure returned from the `checker` package.
+
+### Domain results
+
+Here's a quick synopsis of the top-level fields:
+
+ - `domain`: the domain name that the scan was performed on.
+ - `status`: Whether the check succeeded overall, and some more specific common failure types. Types 4-6 are types of test failures that are particularly common.
+    - 0: Success, all TLS tests passed.
+    - 1: Warning, at least one TLS test produced a warning.
+    - 2: Failure, at least one TLS test failed.
+    - 3: Error, something went wrong during the test.
+    - 4: NoSTARTTLS, at least one of your mailboxes did not advertise STARTTLS.
+    - 5: CouldNotConnect, could not connect to any mailbox.
+    - 6: BadHostnameFailure, one of your mailbox's provided certificates didn't match its hostname.
+ - `message`: A more detailed description of the failure type.
+ - `preferred_hostnames`: A misnomer, but refers to mailboxes that passed the connectivity test.
+ - `mta_sts`: result for MTA STS check.
+ - `extra_results`: A map of other security checks for this domain.
+ - `results`: A map of mailbox hostnames to their individual results.
+
+### Hostname results
+
+Here's a sample, 
+```
 {
-  "domain": "example.com"
+    "status": 0,
+    "checks": {
+        "connectivity": { "status": 0 },
+        "certificate": {
+            "status": 2,
+            "messages": ["Hostname doesn't match any name in certificate",
+                         "Certificate root is not trusted"]
+        },
+        "starttls": { "status": 0 },
+        "version": { "status": 0 },
+    }
 }
 ```
+
+ - `checks`: A result can have a suite of checks. `checks` is a map from a particular check name to its result.
+ - `status`: The status of a particular check, or the overall suite. Can be 0 through 3, which are `Success`, `Warning`, `Failure`, `Error`. The overall suite status takes the max status of all the sub-checks.
```

Comment:
Above, you define 7 potential values of `status`, but here you only identify 4 of them (maybe these two sections got out of sync with one another?).

Reply:
Oops, now I see that these are for two different kinds of replies!